### ISSUE TITLE
AdvancedPOP3Client doesn't notice SSL is secure

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#1809]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#1809|
|Type|defect|
|Created|2006-06-09 20:58:39Z|

Attempting a plaintext login over SSL with AdvancedPOP3Client will raise an InsecureAuthenticationDisallowed error.  This shouldn't happen.

<details><summary>Searchable metadata</summary>

```
trac-id__1809 1809
type__defect defect
reporter__exarkun exarkun
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__mail mail
keywords__ 
time__1149886719000000 1149886719000000
changetime__1150001322000000 1150001322000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

Ready for review in [branches/secure-pop3-login-1809](branches/secure-pop3-login-1809)

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)|@dreid set owner to @dreid|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)|@dreid set owner to @exarkun|
|-|-|

This looks good.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set status to `closed`|
|-|-|

(In [17172]) Merge secure-pop3-login-1809

Author: exarkun
Reviewer: dreid
Fixes [#1809](https://github.com/twisted/twisted/issues/1809)

Allow plaintext logins over SSL or TLS.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
